web/pdf_viewer.js

```diff
@@ -1650,6 +1654,43 @@ class PDFViewer {
     });
   }
 
+  #switchToEditAnnotationMode() {
+    const visible = this._getVisiblePages();
+    const pagesToRefresh = [];
+    const { first, ids, last, views } = visible;
+    let keepFirst = false;
+    let keepLast = false;
+    for (const page of views) {
+      const { view } = page;
+      if (!view.hasEditableAnnotations()) {
+        ids.delete(view.id);
+        continue;
+      }
+      pagesToRefresh.push(page);
+      if (page === first) {
+        keepFirst = true;
+      }
+      if (page === last) {
+        keepLast = true;
+      }
+    }
+
+    if (pagesToRefresh.length === 0) {
+      return null;
+    }
+    if (!keepFirst) {
+      visible.first = pagesToRefresh[0];
+    }
+    if (!keepLast) {
+      visible.last = pagesToRefresh.at(-1);
+    }
+    visible.views = pagesToRefresh;
+
+    this.renderingQueue.renderHighestPriority(visible);
```

Comment:
Nit: I'd probably just write this as follows, and thus be able to simplify/shorten this method overall.
```suggestion
    this.renderingQueue.renderHighestPriority({
      first: pagesToRefresh[0];
      last: pagesToRefresh.at(-1),
      views: pagesToRefresh,
      ids,
    });
```